flash_attn/models/gpt.py

```diff
@@ -824,14 +832,6 @@ def combine_qkv_headdim(state_dicts, state_dict, key):
                 ]
                 state_dict[key] = rearrange(torch.cat(xs, dim=1), "three d ... -> (three d) ...")
             else:
-                xs = [
```

Comment:
sorry i was a bit sloppy when i refactored this part. partially this was due to there not being a test. 

would you mind adding a test for the GQA and MQA case?

Reply:
e.g., i had a test that i ran for standard MHA which i never pushed, which 1) shards a state_dict, 2) merges the shards back, and 3) compares the original vs recovered.

Reply:
added test :)